### PR TITLE
Separate tomcat backend from print proxy

### PR DESCRIPTION
**Pull request by procrastinatio**
Thsi PR add the possibility to split the print proxy (pylons wsgi) from the tomcat print backend, to potentially host both of them on separate host (mapfish print war is not maintained anymore and won't run on debain jessie)

http://service-print.dev.bgdi.ch/mom_tomcat_index/

Nota: test fails because for now this project still runs on wheezy and jenkins runs on jessie

### CODE REVIEW

**Comment by procrastinatio** [comment]
Someone to review @gjn @loicgasser 

And where to put the backend ? Beanstalk ? Docker ? print v4 ? No backend ?

File: rc_dev [hunk 1]
```diff
@@ -3,7 +3,8 @@ export API_URL=//mf-chsdi3.dev.bgdi.ch
 export PRINT_PROXY_URL=//service-print.dev.bgdi.ch
 export DEPLOY_TARGET=dev
 export MODWSGI_CONFIG=production.ini
-export PRINT_WAR=main
+export PRINT_WAR=mom_tomcat_index
```
Comment (loicgasser) [inline rc_dev]: Keep main

File: rc_dev [hunk 2]
```diff
@@ -3,7 +3,8 @@ export API_URL=//mf-chsdi3.dev.bgdi.ch
...
 export ROBOTS_FILE=robots.txt
 export SERVER_PORT=6543
 export WSGI_THREADS=15
+export TOMCAT_BASE_URL=http://localhost:8080
```
Comment (loicgasser) [inline rc_dev]: Is that needed ?

File: tomcat/index.html [hunk 3]
```diff
@@ -1,231 +1,82 @@
...
-                error: function (data) {
-                    $('#messages').text('Error creating report: ' + data.statusText);
-                    disableUI(false);
-                },
-                dataType: 'application/json'
+  <head>
+      <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.4.2/jquery.min.js"></script>
+      <script type="text/javascript">
+          function post() {
+          var data = $('#spec').serialize(true);
```
Comment (loicgasser) [inline tomcat/index.html]: Indent

File: apache/tomcat-print.conf.in [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<Proxy ajp://localhost:8009/service-print-${print_war}>
+<Proxy  ${tomcat_base_url}/service-print-${print_war} >
```
Comment (loicgasser) [inline apache/tomcat-print.conf.in]: White space at the end

**Review by loicgasser — COMMENTED** [review]
Great! A good first step. Some small comments only.
Regarding Jessie, I think the problem comes from Jenkins instance configuration.

```
32mUpdating print war...[m
[32mPrint war creation was successful.[m
[32mRemoving temp directory[m
[32mRestarting tomcat...[m
jenkins is not in the sudoers file.  This incident will be reported.
```

Build should work now on Jessie as well.

**Comment by gjn** [comment]
Is docker an option? We have working rancher infrastructure that we could use to manage deploys of it, DNS, etc.

**Comment by procrastinatio** [comment]
Anything is an options, as long as you have a servlet engine (tomcat 7) and a fixed hostname. A *.war* file are already a _container_ for that purpose, putting a whole linux distro in a docker container only to serve a servlet is in my view stupid econamically speaking, but  cool and moderne.

**Comment by gjn** [comment]
1. Create Beanstalk stack (dev/int on demand, production, aws-admin account), managable with make
2. Create test version of print-proxy in jessie and beanstalk backend
3. Go Live

Goal: have testlink for stakeholders end of next week.

**Pull request merged**